core_test/carmirror_test.go

```diff
@@ -321,29 +338,30 @@ func MockBatchTransfer(sender_store *mock.Store, receiver_store *mock.Store, roo
 	snapshotAfter := stats.GLOBAL_REPORTING.Snapshot()
 	diff := snapshotBefore.Diff(snapshotAfter)
 	diff.Write(&log.SugaredLogger)
-	if diagram {
-		// Write the state diagrammer to a file
-		f, err := os.OpenFile(stateDiagramsFile, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
-		if err != nil {
-			panic(err)
-		}
-		defer f.Close()
-		batchSourceOrchestratorDiagrammer.Close()
-		batchSinkOrchestratorDiagrammer.Close()
-		batchSourceOrchestratorDiagrammer.Write(f)
-		batchSinkOrchestratorDiagrammer.Write(f)
-	}
 	return nil
 }
 
 func TestMockTransferToEmptyStoreSingleBatchNoDelay(t *testing.T) {
 	senderStore := mock.NewStore(mock.DefaultConfig())
 	root := mock.AddRandomTree(context.Background(), senderStore, 10, 5, 0.0)
 	receiverStore := mock.NewStore(mock.DefaultConfig())
-	MockBatchTransfer(senderStore, receiverStore, root, 5000, 0, 0, true)
+
+	// Set up diagrammers
+	w := diagramWriterForTest(t)
```

Comment:
I have to set up the diagrammers and writer here so I can close afterward.  Adds a little extra noise for cases where you want the diagrams, but keeps it cleaner for cases where you don't.